.github/actions/download-with-retry/action.yml

```
@@ -0,0 +1,37 @@
+name: Download with retry
+description: Downloads a file with retry logic (3 attempts, 20s delay)
+
+inputs:
+  url:
+    description: URL to download from
+    required: true
+  output:
+    description: Output filename
+    required: true
+  user-agent:
+    description: User agent string
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Download with retry
+      shell: bash
+      run: |
+        for attempt in 1 2 3; do
```

Please use playful item names.
```suggestion
        for attempt in john john final; do
```